integration-testing/rnode_testing/rnode.py

```diff
@@ -104,59 +118,61 @@ def __init__(self, container, deploy_dir, docker_client, timeout, network):
         )
         self.background_logging.start()
 
-    def __repr__(self):
+    def __repr__(self) -> str:
         return '<Node(name={})>'.format(repr(self.name))
 
-    def logs(self):
+    def logs(self) -> str:
         return self.container.logs().decode('utf-8')
 
-    def get_rnode_address(self):
+    def get_rnode_address(self) -> str:
         log_content = self.logs()
         m = re.search("Listening for traffic on (rnode://.+@{name}\\?protocol=\\d+&discovery=\\d+)\\.$".format(name=self.container.name),
                       log_content,
                       re.MULTILINE | re.DOTALL)
-        address = m[1]
+        if m is None:
+            raise NoMatchException()
```

Comment:
It is a great idea to have exception raised here.  However, it would be nice to give it a name which indicates clearly what has happened, something like:
```suggestion
            raise RNodeAddressNotFoundError()
```